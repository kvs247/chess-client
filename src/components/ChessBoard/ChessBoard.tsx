import { FC } from "react";
import { fenToPiecesArray } from "../../helpers/fenHelper";

const lightSquare = "white";
const darkSquare = "green";

interface ChessBoardParams {
  fen: string;
}

const ChessBoard: FC<ChessBoardParams> = ({ fen }) => {
  const piecesArray = fenToPiecesArray(fen);

  // Create squares
  const squares = [];
  for (let i = 0; i < 64; i++) {
    const row = Math.floor(i / 8);
    const color = (
      (((row % 2 === 0) && (i % 2 === 0)) ||
        ((row % 2 === 1) && (i % 2) === 1)) 
          ? lightSquare : darkSquare
    );

    squares.push(
      <div
        key={i}
        style={{
          backgroundColor: color,
          height: "100%",
          width: "100%",
        }}
      >
        {piecesArray[i] ?? ""}
      </div>
    );

  }

  return (
    <div
      style={{
        backgroundColor: "red",
        width: "calc(min(90vh, 90vw))",
        height: "calc(min(90vh, 90vw))",
        display: "grid",
        gridTemplateColumns: "repeat(8, 1fr)",
        // gridTemplateRows: "repeat(8, 1fr)",
        // position: "relative",
      }}
    >
      {squares}
    </div>
  );
};

export default ChessBoard;