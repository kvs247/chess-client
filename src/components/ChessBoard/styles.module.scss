.Square {
  height: 100%;
  width: 100%;
}

.Wrapper {
  height: calc(min(90vh, 90vw));
  width: calc(min(90vh, 90vw));
  display: grid;
  grid-template-columns: repeat(8, 1fr);
  grid-template-rows: repeat(8, 1fr);
}

.PieceImg {
  max-width: 100%;
  max-height: 100%;
}