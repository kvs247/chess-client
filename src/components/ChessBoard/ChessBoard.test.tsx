import "@testing-library/jest-dom";
import { render, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import ChessBoard from "./ChessBoard";
import startingFEN from "../../startingFEN";

afterEach(cleanup);

it("renders", () => {
  const initialState = {};

  const mockStore = configureStore();
  const store = mockStore(initialState);

  const { asFragment } = render(
    <Provider store={store}>
      <ChessBoard fen={startingFEN} />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});