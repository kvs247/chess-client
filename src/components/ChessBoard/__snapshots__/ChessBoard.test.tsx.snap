// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div
  style="background-color:red;width:calc(min(90vh, 90vw));height:calc(min(90vh, 90vw));display:grid;grid-template-columns:repeat(8, 1fr)"
>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    r
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    n
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    b
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    q
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    k
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    b
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    n
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    r
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    p
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:green;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  />
  <div
    style="background-color:white;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    P
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    R
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    N
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    B
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    Q
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    K
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    B
  </div>
  <div
    style="background-color:green;height:100%;width:100%"
  >
    N
  </div>
  <div
    style="background-color:white;height:100%;width:100%"
  >
    R
  </div>
</div>
`;
